debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSnapshotChangeEventSource.java

```diff
@@ -143,26 +150,178 @@ protected void determineSnapshotOffset(RelationalSnapshotContext<OraclePartition
+                    // Got log that contains the current SCN; this is the one we want to mine.
+                    String addLogFileStatement = SqlUtils.addLogFileStatement("DBMS_LOGMNR.ADDFILE", logFile.getFileName());
+                    connection.executeWithoutCommitting(addLogFileStatement);
+
+                    String logMinerStart = "BEGIN sys.dbms_logmnr.start_logmnr("
+                            + "OPTIONS => DBMS_LOGMNR.DICT_FROM_ONLINE_CATALOG + DBMS_LOGMNR.NO_ROWID_IN_STMT);"
+                            + "END;";
+                    connection.executeWithoutCommitting(logMinerStart);
+
+                    final long currentScnValue = currentScn.longValue();
+                    try {
+                        connection.call("SELECT SCN, START_SCN, COMMIT_SCN, OPERATION, XID FROM V$LOGMNR_CONTENTS WHERE SCN = ?", s -> {
+                            s.setLong(1, currentScnValue);
+                        }, rs -> {
+                            if (rs.next()) {
+                                LOGGER.trace("\tCurrent SCN is operation '{}'", rs.getString("OPERATION"));
+                                final String transactionId = HexConverter.convertToHexString(rs.getBytes("XID"));
+                                if (!snapshotPendingTransactions.containsKey(transactionId)) {
+                                    // in this case we need to adjust the currentScn.
+                                    final String startScnStr = rs.getString("START_SCN");
+                                    if (!Strings.isNullOrEmpty(startScnStr)) {
+                                        // We were able to resolve the start of the transaction.
+                                        // Make sure pending transactions include this one.
+                                        snapshotPendingTransactions.put(transactionId, Scn.valueOf(startScnStr));
+                                    }
+                                    else if (rs.getString("OPERATION").equals("START")) {
+                                        // We were able to resolve the start of the transaction.
+                                        // Make sure pending transactions include this one.
+                                        snapshotPendingTransactions.put(transactionId, Scn.valueOf(rs.getString("SCN")).subtract(Scn.ONE));
+                                    }
+                                    else {
+                                        // Unable to resolve when the transaction started, log warning
+                                        LOGGER.warn("Unable to resolve start SCN for transaction {}, transaction will not be included.", transactionId);
+                                    }
+                                }
+                            }
+                            else {
+                                LOGGER.warn("Failed to find entry for current SCN {}, associated transaction will likely not be included.", currentScnValue);
```

Comment:
This seems to be logged in almost all tests, need to see if there is a better way to avoid this.  This could be because the current SCN may not have been flushed to the redo logs and if so, we may need to find a more appropriate way to just resolve the snapshot scn entirely than based solely on the current SCN.